src/SFA.DAS.ApplyService.Application/Apply/Gateway/UpdateGatewayReviewStatusAndCommentCommandHandler.cs

```diff
@@ -59,6 +63,8 @@ public async Task<Unit> Handle(UpdateGatewayReviewStatusAndCommentCommand reques
                 _auditService.AuditInsert(oversightReview);
                 _oversightReviewRepository.Add(oversightReview);
                 _auditService.Save();
+
+                await _applicationUpdatedEmailService.SendEmail(request.ApplicationId);
```

Comment:
As it stands, this email will be sent and subsequently the database transaction will be committed. In the event of a db connection failure or similar, we'd send the email without having made a state change. It might be worth forcing the commit immediately before this line, so that this can never happen.

To do so, inject the `IUnitOfWork` dependency into the handler, and call its `Commit` method immediately before `SendEmail`.

Happy to discuss.